service/s3/s3crypto/s3crypto_client.go

```diff
@@ -0,0 +1,175 @@
+package s3crypto
+
+import (
+	"encoding/hex"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+// Client is an S3 crypto client. By default we will use EncryptionOnly mode which
+// will use AES Wrap for key wrapping and AES GCM for content encryption.
+// AES GCM will load all data into memory. However, the rest of the content algorithms
+// do not load the entire contents into memory.
+type Client struct {
+	S3     *s3.S3
```

Comment:
Can the S3 be the `s3iface.S3API` interface instead? this will allow users to mock out the SDK's low level service client.